docs/standard/serialization/system-text-json-how-to.md

```diff
@@ -0,0 +1,768 @@
+
+```csharp
+class WeatherForecast
+{
+    public DateTimeOffset Date { get; set; }
+    public int TemperatureC { get; set; }
+    public string Summary { get; set; }
+    [JsonExtensionData]
+    public Dictionary<string, object> ExtensionData { get; set; }
+}
+```
+
+When you deserialize the JSON shown earlier into this sample type, the extra data becomes key-value pairs of the `ExtensionData` property:
+
+|Property |Value  |Notes  |
+|---------|---------|---------|
+| Date    | 8/1/2019 12:00:00 AM -07:00||
+| TemperatureC| 0 | Case-sensitive mismatch (`temperatureC` in the JSON), so the property isn't set. |
+| Summary | Hot ||
+| ExtensionData | temperatureC: 25 |Since the case didn't match, this JSON property is an extra and becomes a key-value pair in the dictionary.|
+|| DatesAvailable:<br>  8/1/2019 12:00:00 AM -07:00<br>8/2/2019 12:00:00 AM -07:00 |Extra property from the JSON becomes a key-value pair, with an array as the value object.|
+| |SummaryWords:<br>Cool<br>Windy<br>Humid |Extra property from the JSON becomes a key-value pair, with an array as the value object.|
+
+When the target object is serialized, the extension data key value pairs become JSON properties just as they were in the incoming JSON:
+
+```json
+{
+  "Date": "2019-08-01T00:00:00-07:00",
+  "TemperatureC": 0,
+  "Summary": "Hot",
+  "temperatureC": 25,
+  "DatesAvailable": [
+    "2019-08-01T00:00:00-07:00",
+    "2019-08-02T00:00:00-07:00"
+  ],
+  "SummaryWords": [
+    "Cool",
+    "Windy",
+    "Humid"
+  ]
+}
+```
+
+Notice that the `ExtensionData` property name doesn't appear in the JSON. This behavior lets you round-trip the JSON without losing any extra data that otherwise wouldn't be deserialized.
+
+## Use Utf8JsonWriter directly
+
+The following example shows how to use the <xref:System.Text.Json.Utf8JsonWriter> class directly.
+
+```csharp
+var options = new JsonWriterOptions
+{
+    Indented = true
+};
+
+using (var stream = new MemoryStream())
+{
+    using (var writer = new Utf8JsonWriter(stream, options))
+    {
+        writer.WriteStartObject();
+        writer.WriteString("date", DateTimeOffset.UtcNow);
+        writer.WriteNumber("temp", 42);
+        writer.WriteEndObject();
+    }
+
+    string json = Encoding.UTF8.GetString(stream.ToArray());
+    Console.WriteLine(json);
+}
+```
+
+## Use Utf8JsonReader directly
+
+The following example shows how to use the <xref:System.Text.Json.Utf8JsonReader> class directly. The code assumes that the `jsonUtf8` variable is a byte array that contains valid JSON.
+
+```csharp
+Utf8JsonReader reader = new Utf8JsonReader(jsonUtf8, isFinalBlock: true, state: default);
+
+while (reader.Read())
```

Comment:
Similar to writing, a few other interesting scenarios come to mind that are worth adding samples for:
1) Reading really large JSON file - say 3 GB.
2) Reading asynchronously (using the state for continuation).
3) Customize the escaping/encoder behavior when writing non-ascii text.
4) [advanced/can be deferred for now] Reading to a System.IO.Pipeline.PipeReader

And of course the custom options (comment handling - skipping/allow, trailing commas, max depth)

Some material from the initial design that might help (contains prototype code/some re-entrancy and async snippets related to the reader):
https://github.com/dotnet/apireviews/blob/master/2018/System.Text.Json/Additional.Notes.md

Again, I am just adding that link as context and I wouldn't recommend reading too much into it since its likely out-dated or overly complex. I can help provide cleaner samples later.